src/Groups/GrpAb.jl

```diff
@@ -12,3 +12,348 @@ function restrict_codomain(f::GrpAbFinGenMap)
+
+
+################################################################################
+#
+#   Conjugacy Classes in GrpAbFinGen groups: just wrap elements
+#
+################################################################################
+
+struct GrpAbFinGenConjClass{T<:GrpAbFinGen, S<:Union{GrpAbFinGenElem,GrpAbFinGen}} <: GroupConjClass{T, S}
+   X::T
+   repr::S
+end
+
+function Base.hash(C::GrpAbFinGenConjClass, h::UInt)
+  return Base.hash(Representative(C), h)
+end
+
+function Base.show(io::IO, C::GrpAbFinGenConjClass)
+  print(io, string(representative(C)), " ^ ", string(C.X))
+end
+
+==(a::GrpAbFinGenConjClass, b::GrpAbFinGenConjClass) = representative(a) == representative(b)
+
+Base.length(::Type{T}, C::GrpAbFinGenConjClass) where T <: IntegerUnion = T(1)
+
+Base.length(C::GrpAbFinGenConjClass) = ZZRingElem(1)
+
+
+################################################################################
+#
+#   Conjugacy classes of elements
+#
+################################################################################
+
+conjugacy_class(G::GrpAbFinGen, g::GrpAbFinGenElem) = GrpAbFinGenConjClass(G, g)
+
+Base.eltype(::Type{GrpAbFinGenConjClass{T,S}}) where {T,S} = S
+
+Base.rand(C::GrpAbFinGenConjClass) = representative(C)
+
+Base.rand(rng::Random.AbstractRNG, C::GrpAbFinGenConjClass) = representative(C)
+
+number_conjugacy_classes(G::GrpAbFinGen) = ZZRingElem(order(G))
+
+number_conjugacy_classes(::Type{T}, G::GrpAbFinGen) where T <: IntegerUnion = T(order(G))
+
+conjugacy_classes(G::GrpAbFinGen) = [GrpAbFinGenConjClass(G, x) for x in G]
+
+is_conjugate(G::GrpAbFinGen, x::GrpAbFinGenElem, y::GrpAbFinGenElem) = (x == y)
+
+function representative_action(G::GrpAbFinGen, x::GrpAbFinGenElem, y::GrpAbFinGenElem)
+   x == y ? (true, zero(G)) : (false, nothing)
+end
+
+
+################################################################################
+#
+#   Conjugacy classes of subgroups
+#
+################################################################################
+
+conjugacy_class(G::T, H::T) where T <: GrpAbFinGen = GrpAbFinGenConjClass(G, H)
+
+function conjugacy_classes_subgroups(G::GrpAbFinGen)
+   return [conjugacy_class(G, H) for (H, mp) in subgroups(G)]
+end
+
+#TODO: Does Hecke provide the following?
+# subgroup_reps(G::GrpAbFinGen; order::ZZRingElem = ZZRingElem(-1))
+# conjugacy_classes_maximal_subgroups(G::GrpAbFinGen)
+# maximal_subgroup_reps(G::GrpAbFinGen)
+# low_index_subgroup_reps(G::GrpAbFinGen, n::Int)
+
+conjugate_group(G::GrpAbFinGen, x::GrpAbFinGenElem) = G
+
+Base.:^(G::GrpAbFinGen, x::GrpAbFinGenElem) = G
+
+function is_conjugate(G::GrpAbFinGen, H::GrpAbFinGen, K::GrpAbFinGen)
+   if is_subgroup(H, K)[1] && is_subgroup(K, H)[1]
+     return true
+   else
+     return false
+   end
+end
+
+function representative_action(G::GrpAbFinGen, H::GrpAbFinGen, K::GrpAbFinGen)
+   if is_subgroup(H, K)[1] && is_subgroup(K, H)[1]
+     return true, zero(G)
+   else
+     return false, nothing
```

Comment:
```suggestion
     return false, zero(G)
```
is there harm in doing that?

Reply:
Currently the `representative_action` documentation promises `nothing`, and the method for `GAPGroup` does this.
We can change the definition and the methods such that always the identity element of the group is returned instead of `nothing`.